File: apps/api/src/config/app.config.ts
import { registerAs } from '@nestjs/config';
import { environment } from './environment.config';

export interface AppConfig {
  host?: string;
  port: number;
  isDevelopmentEnvironment: boolean;
  isProductionEnvironment: boolean;
  isDocumentationEnabled: boolean;
}

export const appConfig = registerAs('app', (): AppConfig => {
  const nodeEnvironment = environment.get('NODE_ENV', 'development');
  const isProductionEnvironment = nodeEnvironment === 'production';

  return {
    host: environment.get('HOST', 'localhost'),
    port: environment.getNumber('PORT', 4000),
    isDevelopmentEnvironment: !isProductionEnvironment,
    isProductionEnvironment: isProductionEnvironment,
    isDocumentationEnabled: environment.getBoolean('DOCUMENTATION_ENABLED', false),
  };
});

File: apps/cms/src/Config/AppConfig.cs
namespace GameGuild.Config;

public class AppConfig
{
    public string Host
    {
        get;
        set;
    } = "localhost";

    public int Port
    {
        get;
        set;
    } = 5000;

    public bool IsDevelopmentEnvironment
    {
        get;
        set;
    }

    public bool IsProductionEnvironment
    {
        get;
        set;
    }

    public bool IsDocumentationEnabled
    {
        get;
        set;
    }

    public string Environment
    {
        get;
        set;
    } = "Development";
}
